pinot-broker/src/main/java/org/apache/pinot/broker/broker/helix/HelixBrokerStarter.java

```diff
@@ -237,16 +238,30 @@ public void start()
         .addPreConnectCallback(() -> brokerMetrics.addMeteredGlobalValue(BrokerMeter.HELIX_ZOOKEEPER_RECONNECTS, 1L));
 
     // Register the service status handler
-    LOGGER.info("Registering service status handler");
+    registerServiceStatusHandler();
+
+    LOGGER.info("Finish starting Pinot broker");
+  }
+
+  /**
+   * Fetches the resources to monitor and registers the {@link org.apache.pinot.common.utils.ServiceStatus.ServiceStatusCallback}s
+   */
+  private void registerServiceStatusHandler() {
+    List<String> resourcesToMonitor = new ArrayList<>(1);
+    IdealState resourceIdealState = _helixAdmin.getResourceIdealState(_clusterName, Helix.BROKER_RESOURCE_INSTANCE);
+    if (resourceIdealState != null && resourceIdealState.isEnabled()) {
```

Comment:
dont we also need to check if this broker instance is on for any of the partitions?

Reply:
Hmm yes, fixed. 
Follow up question: We still want to set the ServiceStatusHandlers, even if resourcesToMonitor is empty right? That way the health checks will get response GOOD instead of having no service handlers?

Reply:
If each service handler we want to have end up being empty, then yes. we do need to set one that always returns GOOD. Perhaps that can be the default setting for ServiceStatus.getServiceStatus ?